inference-engine/src/gna_plugin/frontend/scale_factor_calc.hpp

```diff
@@ -454,16 +455,17 @@ class ScaleFactorPerLayer<InferenceEngine::ConcatLayer*> {
         static std::map<std::string, size_t> restarted_counter;
         auto restartedCountIt = restarted_counter.find(concatLayer->name);
         if (restartedCountIt == restarted_counter.end()) {
-            auto pos = restarted_counter.insert({ "concatLayer->name", 0 });
+            auto pos = restarted_counter.insert({ concatLayer->name, 0 });
             restartedCountIt = pos.first;
         }
 
-        if (restartedCountIt->second % 2 == 1) {
+        if (((restartedCountIt->second) / 2) % 2 == 1) {
```

Comment:
we should swap inputs every 2 iterations to make it possible to check if everything is fine after one swap